tests/func/test_run.py

```diff
@@ -948,3 +949,35 @@ def test(self):
 
                     mock_run.assert_called_once()
                     mock_checkout.assert_not_called()
+
+
+class TestPersistentOutput(TestDvc):
+    def test_ignore_build_cache(self):
+        warning = "Build cache is ignored when persisting outputs."
+
+        with open("immutable", "w") as fobj:
+            fobj.write("1")
+
+        cmd = [
+            "run",
+            "--overwrite-dvcfile",
+            "--deps",
+            "immutable",
+            "--outs-persist",
+            "greetings",
+            "echo hello>>greetings",
+        ]
+
+        with self._caplog.at_level(logging.WARNING, logger="dvc"):
+            assert main(cmd) == 0
+            assert warning in self._caplog.text
```

Comment:
There shouldn't be a warning during the first run.

Reply:
@efiop , should we skip the _overwrite stage file_ confirmation when `outs-persist` is used?

Reply:
@mroutis why would we want that though? :slightly_smiling_face: